parse/ValueRefParser.h

```diff
@@ -1,38 +1,329 @@
 #ifndef _ValueRefParser_h_
 #define _ValueRefParser_h_
 
-#include "Lexer.h"
-#include "ParseImpl.h"
+#include "EnumParser.h"
 
 #include "../universe/ValueRefFwd.h"
 #include "../universe/EnumsFwd.h"
 
 #include <boost/spirit/include/qi.hpp>
 
+namespace Condition {
+    struct ConditionBase;
+}
 
 namespace parse {
+    // TODO: Investigate refactoring ValueRef to use variant,
+    // for increased locality of reference.
+    template <typename T>
+    using value_ref_signature = ValueRef::ValueRefBase<T>* ();
+    template <typename T>
+    using value_ref_rule = detail::rule<value_ref_signature<T>>;
+    template <typename T>
+    using value_ref_grammar = detail::grammar<value_ref_signature<T>>;
+}
+
+namespace parse { namespace detail {
+
+    using condition_signature      = Condition::ConditionBase* ();
+    using condition_parser_rule    = rule<condition_signature>;
+    using condition_parser_grammar = grammar<condition_signature>;
+
+using name_token_rule = rule<const char* ()>;
```

Comment:
The inconsistent indentation in this file is mildly confusing.

Reply:
I fixed the indentation.